### PR TITLE
Port for Lexus ESH 2018

### PR DESCRIPTION
**Pull request by TheInventorMan**
Port for 2018 Lexus ES Hybrid models. Testing was done on an ES300h. 

Improved brake detection/disengage by logically ORing redundant BRAKE_PRESSED signals from both BRAKE_MODULE2 and original BRAKE_MODULE messages.

**Checklist**
- [x] added to README
- [x] test route added to [test_car_models](../../selfdrive/test/test_car_models.py)
- [x] route with openpilot: https://my.comma.ai/useradmin/?onebox=37041c500fd30100%7C2020-12-30--12-17-24
- [x] route with stock system: https://my.comma.ai/useradmin/?onebox=37041c500fd30100%7C2020-12-26--15-56-23

### CODE REVIEW

File: selfdrive/car/toyota/values.py [hunk 1]
```diff
@@ -1152,6 +1157,23 @@ class CAR:
       b'\x028646F3304100\x00\x00\x00\x008646G2601200\x00\x00\x00\x00',
     ],
   },
+  CAR.LEXUS_ESH: {
```
Comment (adeebshihadeh) [inline selfdrive/car/toyota/values.py:1160]: add CAR.LEXUS_ESH to IGNORED_FINGERPRINTS

File: README.md [hunk 2]
```diff
@@ -88,6 +88,7 @@ Supported Cars
 | Lexus     | CT Hybrid 2017-18             | LSS               | Stock<sup>3</sup>| 0mph               | 0mph              |
 | Lexus     | ES 2019-20                    | All               | openpilot        | 0mph               | 0mph              |
 | Lexus     | ES Hybrid 2019                | All               | openpilot        | 0mph               | 0mph              |
+| Lexus     | ES Hybrid 2016-18             | LSS               | Stock<sup>3</sup>| 0mph               | 0mph              |
```
Comment (adeebshihadeh) [inline README.md:91]: let's only put 2018 since 2016-2017 haven't been tested

File: selfdrive/car/toyota/values.py (hunk 1 above)
Comment (TheInventorMan) [reply selfdrive/car/toyota/values.py:1160]: done!

File: README.md (hunk 2 above)
Comment (TheInventorMan) [reply README.md]: sure thing, fixed

**Comment by TheInventorMan** [comment]
Looks like the 'test car models' CI test is failing since the route is not being found. I made the drive public on the useradmin console--am I missing a step somewhere?

**Comment by adeebshihadeh** [comment]
We have to manually sync the test routes to a special CI bucket. Re-running that test, and should be good to merge once it passes.

File: selfdrive/car/toyota/carstate.py [hunk 3]
```diff
@@ -29,7 +29,7 @@ def update(self, cp, cp_cam):
                         cp.vl["SEATS_DOORS"]['DOOR_OPEN_RL'], cp.vl["SEATS_DOORS"]['DOOR_OPEN_RR']])
     ret.seatbeltUnlatched = cp.vl["SEATS_DOORS"]['SEATBELT_DRIVER_UNLATCHED'] != 0
 
-    ret.brakePressed = cp.vl["BRAKE_MODULE"]['BRAKE_PRESSED'] != 0
+    ret.brakePressed = min(cp.vl["BRAKE_MODULE"]['BRAKE_PRESSED'] + cp.vl["BRAKE_MODULE2"]['BRAKE_PRESSED'], 1) != 0
```
Comment (adeebshihadeh) [inline selfdrive/car/toyota/carstate.py:32]: is the exiting signal not correct for the ES? if not, we should change the DBC so the signal name is correct

Comment (TheInventorMan) [reply selfdrive/car/toyota/carstate.py]: Ah, sorry, forgot to add this to the PR description. It seems like BRAKE_MODULE2->BRAKE_PRESSED is a redundant signal, and ORing this with the original BRAKE_MODULE signal should make things safer. I've made a little change to improve readability.

Comment (TheInventorMan) [reply selfdrive/car/toyota/carstate.py]: The existing signal is correct and the two match up 999,999 times out of 1,000,000. There was this one time about a year ago when OP did NOT disengage upon brake press, hence the fix.

Comment (adeebshihadeh) [reply selfdrive/car/toyota/carstate.py]: do you have a route where this happened?

Comment (adeebshihadeh) [reply selfdrive/car/toyota/carstate.py]: In any case, this change should be moved to a separate PR. The car port is good to merge once this is split out.

Comment (TheInventorMan) [reply selfdrive/car/toyota/carstate.py]: Sounds good, I'll remove this change for now and open a separate PR for it

**Comment by TheInventorMan** [comment]
@adeebshihadeh I think we're good to go now!

File: README.md (hunk 2 above)
Comment (way129) [reply README.md]: I'll test on es300h 2017

**Pull request merged**